src/main/kotlin/de/bund/digitalservice/useid/widget/WidgetController.kt

```diff
@@ -1,31 +1,54 @@
 package de.bund.digitalservice.useid.widget
 
 import de.bund.digitalservice.useid.config.ApplicationProperties
+import de.bund.digitalservice.useid.tracking.TrackingService
+import de.bund.digitalservice.useid.tracking.WidgetTracking
 import io.micrometer.core.annotation.Timed
 import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
 import org.springframework.http.server.reactive.ServerHttpRequest
 import org.springframework.stereotype.Controller
 import org.springframework.ui.Model
 import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PostMapping
 import org.springframework.web.reactive.result.view.Rendering
 
 internal const val WIDGET_PAGE = "widget"
 internal const val INCOMPATIBLE_PAGE = "incompatible"
 internal const val FALLBACK_PAGE = "eID-Client"
+internal const val APP_OPENED = "app-opened"
 
 @Controller
 @Timed
 class WidgetController(
     private val applicationProperties: ApplicationProperties,
-    private val widgetProperties: WidgetProperties
+    private val widgetProperties: WidgetProperties,
+    private val trackingService: TrackingService,
+    private val widgetTracking: WidgetTracking
 ) {
     private val defaultViewHeaderConfig = mapOf(
         "baseUrl" to applicationProperties.baseUrl,
         "metaTag" to widgetProperties.metaTag
     )
 
+    @PostMapping("/$APP_OPENED")
+    fun handleAppOpened(): ResponseEntity<String> {
+        trackingService.sendMatomoEvent(
```

Comment:
I'm not that much into the spring/kotlin details, but is this always working? Don't we have to add this into a background task? Or is that done automatically?

Reply:
What do you mean with "background task"?

Reply:
I have to admit, I might be mixing sth up, but the call to matomo should be asynchronous. I don't remember at the moment how this is handled in Spring.

Reply:
> We should definitely execute the sending event asynchronously and immediately respond with OK. But that might be solved implicitly if the controller is refactored to reactive approach? Is that what you talking about @code28 ?

Yes :)